From b9c23f40cd02cbea5bece333e91c74b5c547efef Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 20 Apr 2022 11:31:52 +0200
Subject: [PATCH 162/954] feat(mtd): add 4-bytes address opcodes for NOR

This patch adds 4-bytes read and fast read opcodes.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4d3a030bbc041a398d49c7bab6e520a364ec4457 feat(mtd): add 4-bytes address opcodes for NOR]
Change-Id: Ic5dbb40f52468c9e912f96637346f0f29cea18a0
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/248752
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 include/drivers/spi_nor.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/drivers/spi_nor.h b/include/drivers/spi_nor.h
index 444e84237b..07dee5a9dc 100644
--- a/include/drivers/spi_nor.h
+++ b/include/drivers/spi_nor.h
@@ -32,6 +32,10 @@
 #define SPI_NOR_OP_READ_1_1_8	0x8BU	/* Read data bytes (Octal Output SPI) */
 #define SPI_NOR_OP_READ_1_8_8	0xCBU	/* Read data bytes (Octal I/O SPI) */
 
+/* 4-bytes address opcodes */
+#define SPI_NOR_OP_READ_4B	0x13U	/* Read data bytes (low frequency) */
+#define SPI_NOR_OP_READ_FAST_4B	0x0CU	/* Read data bytes (high frequency) */
+
 /* Flags for NOR specific configuration */
 #define SPI_NOR_USE_FSR		BIT(0)
 #define SPI_NOR_USE_BANK	BIT(1)
-- 
2.53.0

